### ISSUE TITLE
False positive when name is not the same as the dir

### ISSUE THREAD

**Issue by TimeEngineer** [rust-lang/rust-clippy#10271]
### Summary

Let's say you have a project:

```
api/
|- foo/
    |- src/
        |- lib.rs
    |- Cargo.toml
src/
|- foo.rs
|- lib.rs
Cargo.toml
```

```Toml
# /Cargo.toml
[package]
name = "false_positive"
version = "0.1.0"
edition = "2021"

[workspace]
members = ["api/*"]

[dependencies]
renamed_foo = { path = "api/foo" }
```

```Rust
// /src/lib.rs
mod foo;

pub use foo::*;
pub use renamed_foo::*;
```

```Toml
# /api/foo/Cargo.toml
[package]
name = "renamed_foo" # Not the same name as the dir
version = "0.1.0"
edition = "2021"

[dependencies]
```

```Rust
// /api/foo/src/lib.rs
// Note: This is important that this file is not empty.
pub fn function() {}
```

### Lint Name

clippy::self-named-module-files

### Reproducer

I tried to run with:

```shell
cargo clippy -- --deny=clippy::self_named_module_files
```

I saw this happen:

```shell
error: `mod.rs` files are required, found `src/foo.rs`
 --> src/foo.rs:0:1
  |
  |
  = help: move `src/foo.rs` to `src/foo/mod.rs`
  = note: requested on the command line with `-D clippy::self-named-module-files`

error: could not compile `false_positive` due to previous error
```

I expected to see this happen:
```
no error
```

### Version

```text
rustc 1.66.1 (90743e729 2023-01-10)
binary: rustc
commit-hash: 90743e7298aca107ddaa0c202a4d3604e29bfeb6
commit-date: 2023-01-10
host: x86_64-unknown-linux-gnu
release: 1.66.1
LLVM version: 15.0.2
```
